apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -21,6 +21,19 @@ export default class FieldDropdown extends GoogleBlockly.FieldDropdown {
         }
       };
     }
+
+    // For backwards compatibility, we must account for and format existing blocks
+    // that are missing the second value (used as the generated code for that option).
+    const isMissingSecondValue = menuGenerator.some(
+      entry => entry.length === 1
+    );
+
+    if (isMissingSecondValue) {
+      menuGenerator = menuGenerator.map(entry => {
+        return [entry[0], entry[0]];
+      });
+    }
```

Comment:
I think Ideally we could move a lot of this logic out of the constructor and into their own methods, but when I did that I got an error that I couldn't call `this` before `super()`. Open to ideas, but also ok leaving it here if that ends up making the most sense.